### DISCUSSION TITLE
Using SQLAlchemy with a flaky DB connection

### DISCUSSION THREAD

**Discussion by SDeans0** [sqlalchemy/sqlalchemy#8021] [Usage Questions] [answered]
I am working with a DB which frequently fails on first creating a connection, but when you gain a connection it will persist.

I can gain a connection by wrapping it in a `try/except` block and catching `OperationalError`, retrying until I successfully connect, but this seems a bit inelegant. Using `pre_ping=True` doesn't help, I am still consistently getting errors.

Have use cases like this been considered in SQLAlchemy, and is there a moer direct built in method that I could use?

**Comment by zzzeek** [comment, marked as answer]
you need to write a custom connect routine. your best bet would be writing a do_connect() handler since you get the DBAPI arguments and you can do what you need.  See https://docs.sqlalchemy.org/en/14/core/engines.html#fully-replacing-the-dbapi-connect-function

**Reply by SDeans0** [reply]
Thanks, much appreciated!
